public/pages/CreateMonitor/components/AssociateMonitors/components/MonitorsList.js

```diff
@@ -0,0 +1,286 @@
+      selected[index] = {
+        label: filteredOption[0]?.monitor_name || '',
+        value: monitor.monitor_id,
+      };
+    });
+
+    setSelection(selected);
+    return selected;
+  };
+
+  const generateFields = (selected) => {
+    return _.reduce(
+      Object.keys(selected).length > 1 ? Object.keys(selected) : [0, 1],
+      (result, value, key) => {
+        result.push(key);
+        return result;
+      },
+      []
+    );
+  };
+
+  const monitorsToOptions = (monitors) =>
+    monitors.map((monitor) => ({
+      label: monitor.monitor_name,
+      value: monitor.monitor_id,
+    }));
+
+  const onChange = (options, monitorIdx, form) => {
+    let selected = {
+      ...selection,
+    };
+    if (options[0]) {
+      selected[monitorIdx] = options[0];
+    } else {
+      delete selected[monitorIdx];
+    }
+    setSelection(selected);
+
+    setFormikValues(selected, monitorIdx, form);
+    updateSelection(selected);
+  };
+
+  const onBlur = (e, field, form) => {
+    form.setFieldTouched(formikFieldName, true);
+    form.setFieldTouched(field.name, true);
+  };
+
+  const updateSelection = (selected) => {
+    const newMonitorOptions = [...options];
+    newMonitorOptions.forEach((mon) => {
+      mon.disabled = isSelected(selected, mon);
+    });
+
+    setOptions([...newMonitorOptions]);
+  };
+
+  const setFormikValues = (selected, monitorIdx, form) => {
+    const associatedMonitors = _.get(
+      values,
+      'associatedMonitors',
+      DEFAULT_ASSOCIATED_MONITORS_VALUE
+    );
+    associatedMonitors.sequence.delegates = selectionToFormik(selected);
+    form.setFieldValue(formikValueName, associatedMonitors);
+  };
+
+  const selectionToFormik = (selection) => {
+    const monitors = [];
+    Object.values(selection).forEach((monitor, index) => {
+      monitors.push({
+        order: index + 1,
+        monitor_id: monitor.value,
+      });
+    });
+
+    return monitors;
+  };
+
+  const isSelected = (selected, monitor) => {
+    let isSelected = false;
+    for (const key in selected) {
+      if (selected.hasOwnProperty(key)) {
+        if (selected[key].value === monitor.value) {
+          isSelected = true;
+          break;
+        }
+      }
+    }
+    return isSelected;
+  };
+
+  const onAddMonitor = () => {
+    let nextIndex = Math.max(...fields) + 1;
+    const newMonitorFields = [...fields, nextIndex];
+    setFields(newMonitorFields);
+
+    updateSelection(selection);
+  };
+
+  const onRemoveMonitor = (monitorIdx, idx, form) => {
+    const selected = { ...selection };
+    delete selected[monitorIdx];
+    setSelection(selected);
+
+    const newMonitorFields = [...fields];
+    newMonitorFields.splice(idx, 1);
+    setFields(newMonitorFields);
+
+    setFormikValues(selected, monitorIdx, form);
+    updateSelection(selected);
+  };
+
+  const isValid = () => Object.keys(selection).length > 1;
+
+  const validate = () => {
+    if (!isValid()) return 'Required.';
+  };
```

Comment:
We can use the [required helper validate function](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/public/utils/validate.js#L60)

Reply:
Done